### ISSUE TITLE
Default puller queue is too large

### ISSUE THREAD

**Issue by calmh** [syncthing/syncthing#4910]
Syncthing doesn't like syncing, or maybe new folders, on at least FreeBSD in 0.14.47. But I doubt it's platform dependent really. From the outside, Syncthing just stops responding to outside stimulus. Hitting it with a -QUIT gives tens of thousands of goroutines trying to serve requests.

```
root@pseudo: # grep 'handleRequest' /var/log/messages | head -5
May  1 22:42:13 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x1c4d, 0xc4559165f4, 0x4, 0xc4322a14a0, 0x2d, 0xb40000, 0x20000, 0xc46e585bc0, 0x20, ...)
May  1 22:42:13 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x18fc, 0xc47200a0b8, 0x4, 0xc4664e7e00, 0x2d, 0x16e0000, 0x20000, 0xc4720480c0, 0x20, ...)
May  1 22:42:13 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x16a6, 0xc4457b8ed8, 0x4, 0xc456f78300, 0x2d, 0x620000, 0x20000, 0xc45c2ded20, 0x20, ...)
May  1 22:42:13 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x16ad, 0xc4457b8fec, 0x4, 0xc456f78360, 0x2d, 0x17a0000, 0x20000, 0xc45c2deea0, 0x20, ...)
May  1 22:42:13 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0xa6e, 0xc423fbc3e8, 0x4, 0xc4403601b0, 0x2d, 0xb60000, 0x20000, 0xc43b7f46c0, 0x20, ...)
root@pseudo: # grep 'handleRequest' /var/log/messages | tail -5
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62ed, 0xc52f63a528, 0x4, 0xc531914ff0, 0x2d, 0x1360000, 0x20000, 0xc531a50120, 0x20, ...)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62ef, 0xc52f63a530, 0x4, 0xc531915050, 0x2d, 0x60000, 0x20000, 0xc531a50160, 0x20, ...)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62f2, 0xc52f63a53c, 0x4, 0xc5319150e0, 0x2d, 0x760000, 0x20000, 0xc531a501c0, 0x20, ...)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62f3, 0xc52f63a540, 0x4, 0xc531915110, 0x2d, 0x15a0000, 0x20000, 0xc531a501e0, 0x20, ...)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62f4, 0xc52f63a544, 0x4, 0xc531915140, 0x2d, 0xbc0000, 0x20000, 0xc531a50200, 0x20, ...)
root@pseudo: # grep 'handleRequest' /var/log/messages | wc -l
   21063
root@pseudo: # 
```

Full trace:

```
May  1 22:42:14 pseudo syncthing: goroutine 25730 [runnable]:
May  1 22:42:14 pseudo syncthing: syscall.Syscall6(0x1db, 0x94, 0xc532de8000, 0x20000, 0x11e0000, 0x0, 0x0, 0x20000, 0x20000, 0x0)
May  1 22:42:14 pseudo syncthing:       /usr/local/go/src/syscall/asm_unix_amd64.s:42 +0x5
May  1 22:42:14 pseudo syncthing: syscall.Pread(0x94, 0xc532de8000, 0x20000, 0x20000, 0x11e0000, 0xc532dbf748, 0xc532dbf748, 0x40df78)
May  1 22:42:14 pseudo syncthing: internal/poll.(*FD).Pread(0xc531b09f90, 0xc532de8000, 0x20000, 0x20000, 0x11e0000, 0xd96ac0, 0x429280, 0xc532dbf7a0)
May  1 22:42:14 pseudo syncthing: os.(*File).pread(0xc52cfa2a00, 0xc532de8000, 0x20000, 0x20000, 0x11e0000, 0xc532dbf800, 0x744640, 0xc42036e880)
May  1 22:42:14 pseudo syncthing:       /usr/local/go/src/os/file_unix.go:235 +0x58
May  1 22:42:14 pseudo syncthing: os.(*File).ReadAt(0xc52cfa2a00, 0xc532de8000, 0x20000, 0x20000, 0x11e0000, 0x0, 0x0, 0x0)
May  1 22:42:14 pseudo syncthing:       /usr/local/go/src/os/file.go:125 +0xfd
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/model.readOffsetIntoBuf(0xd99780, 0xc4200678b0, 0xc531914d80, 0x2d, 0x11e0000, 0xc532de8000, 0x20000, 0x20000, 0x0, 0x0)
May  1 22:42:14 pseudo syncthing:       /opt/buildagent/work/src/github.com/syncthing/syncthing/lib/model/model.go:2767 +0x265
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/model.(*Model).Request(0xc4200b0000, 0xc40152a406b67730, 0x6a111ff6b0a6ad45, 0xbe968688a98767c1, 0xcacffc029a6f6a3a, 0xc52f63a4f4, 0x4, 0xc531914d80, 0x2d, 0x11e0000, ...)
May  1 22:42:14 pseudo syncthing:       /opt/buildagent/work/src/github.com/syncthing/syncthing/lib/model/model.go:1379 +0x1133
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62e0, 0xc52f63a4f4, 0x4, 0xc531914d80, 0x2d, 0x11e0000, 0x20000, 0xc52fa29f80, 0x20, ...)
May  1 22:42:14 pseudo syncthing:       /opt/buildagent/work/src/github.com/syncthing/syncthing/lib/protocol/protocol.go:340 +0x619
May  1 22:42:14 pseudo syncthing: syscall.Syscall6(0x1db, 0x668, 0xc532e2e000, 0x20000, 0x480000, 0x0, 0x0, 0x20000, 0x20000, 0x0)
May  1 22:42:14 pseudo syncthing: syscall.Pread(0x668, 0xc532e2e000, 0x20000, 0x20000, 0x480000, 0xc522b85748, 0xc522b85748, 0x40df78)
May  1 22:42:14 pseudo syncthing:       /usr/local/go/src/syscall/zsyscall_freebsd_amd64.go:833 +0x7a
May  1 22:42:14 pseudo syncthing: internal/poll.(*FD).Pread(0xc532e10050, 0xc532e2e000, 0x20000, 0x20000, 0x480000, 0xd96ac0, 0x429280, 0xc522b857a0)
May  1 22:42:14 pseudo syncthing: os.(*File).pread(0xc52cfa2a08, 0xc532e2e000, 0x20000, 0x20000, 0x480000, 0xc522b85800, 0x744640, 0xc42036e880)
May  1 22:42:14 pseudo syncthing:       /usr/local/go/src/os/file_unix.go:235 +0x58
May  1 22:42:14 pseudo syncthing: os.(*File).ReadAt(0xc52cfa2a08, 0xc532e2e000, 0x20000, 0x20000, 0x480000, 0x0, 0x0, 0x0)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/model.readOffsetIntoBuf(0xd99780, 0xc4200678b0, 0xc531914db0, 0x2d, 0x480000, 0xc532e2e000, 0x20000, 0x20000, 0x0, 0x0)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/model.(*Model).Request(0xc4200b0000, 0xc40152a406b67730, 0x6a111ff6b0a6ad45, 0xbe968688a98767c1, 0xcacffc029a6f6a3a, 0xc52f63a4f8, 0x4, 0xc531914db0, 0x2d, 0x480000, ...)
May  1 22:42:14 pseudo syncthing: github.com/syncthing/syncthing/lib/protocol.(*rawConnection).handleRequest(0xc4218860c0, 0x62e1, 0xc52f63a4f8, 0x4, 0xc531914db0, 0x2d, 0x480000, 0x20000, 0xc52fa29fa0, 0x20, ...)
May  1 22:42:14 pseudo syncthing:       /opt/buildagent/work/src/github.com/syncthing/syncthing/lib/protocol/protocol.go:548 +0x1a2
```

I suspect it is something wrong with the new puller limiting system I did.

**Comment by AudriusButkevicius** [comment]
Full trace not so full?

**Comment by AudriusButkevicius** [comment]
Why would this be platform specific?

**Comment by calmh** [comment]
Yeah just full for one of the tens of thousands of identical goroutines. But I found the problem, it's on the requesting side, and thankfully not in 0.14.47.

The other side in my setup runs an imsodin-patched version with the index sorting, which is based on a newer master. That newer master has my large block stuff, wherein I set the default puller size to twice the max block size. But the block size is in bytes and that variable is in kibibytes, so the default puller queue becomes 32 gigs, which exhausts memory...
